src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -90,8 +90,8 @@ private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener
         } else {
             status = new BitbucketBuildStatus(hash, "The tests have started...", "INPROGRESS", url, key, name);
         }
-        new BitbucketChangesetCommentNotifier(bitbucket).buildStatus(status);
-        if (result != null) {
+        if (result != null && !Result.NOT_BUILT.equals(result)) {
```

Comment:
This will never mark the builds as "INPROGRESS".
This is a breaking change for people who depend on the state of a build (like us who have a rule to prevent PR merge until all the builds succeed). If the builds are not marked as "INPROGRESS", then the merge would succeed since bitbucket does not know that a build is running.

Reply:
ah, the reason I moved it was that the message was not printed if result is null, which seems contrary to what you want anyway. I'll move it out.